core/src/main/java/org/apache/iceberg/ManifestFilterManager.java

```diff
@@ -304,6 +309,11 @@ private ManifestFile filterManifest(Schema tableSchema, ManifestFile manifest) {
       return manifest;
     }
 
+    boolean hasDeletedFiles = referencedManifestLocations.contains(manifest.path());
+    if (hasDeletedFiles) {
+      return filterManifestWithDeletedFiles(manifest, tableSchema);
+    }
```

Comment:
referencedManifests are only added to when there's a file that's deleted. If this exists we can skip the `manifestHasDeletedFiles` method which opens up the reader to evaluate if a file has any deletes to begin with. We can skip this because we know absolutely there is at least a single delete! 

We still need to open up the manifest reader and write out the new manifest with the deleted entry, there can be additional criteria for the delete like by filter which need to get evaluated. However, that seems optimal already since we first short circuit evaluate if the path is deleted , and if it is we don't need to evaluate against the partition+metrics.

Reply:
Updated my comment with a possibly more optimized way using positions: https://github.com/apache/iceberg/pull/11131#issuecomment-2350698759

I'll explore this a bit more, but before that I'm going to look into how to effectively benchmark this change so I'm measuring it the right way.